diff --git a/ports/cmake/src/CMakeLists.txt b/ports/cmake/src/CMakeLists.txt
index 6d53ad1..913ca18 100644
--- a/ports/cmake/src/CMakeLists.txt
+++ b/ports/cmake/src/CMakeLists.txt
@@ -74,7 +74,9 @@ check_function_exists(execvp HAVE_EXECVP)
 check_function_exists(ctermid HAVE_CTERMID)
 check_function_exists(clock_gettime HAVE_CLOCK_GETTIME)
 
+set(CMAKE_EXTRA_INCLUDE_FILES "sys/types.h")
 check_type_size(off_t MPG123_SIZEOF_OFF_T)
+unset(CMAKE_EXTRA_INCLUDE_FILES)
 
 if(MPG123_SIZEOF_OFF_T LESS 8)
 check_function_exists(lseek64 LFS_LARGEFILE_64)
